pkg/api/types.go

```diff
@@ -1736,8 +1736,17 @@ type Taint struct {
 	Value string
 	// Required. The effect of the taint on pods
 	// that do not tolerate the taint.
-	// Valid effects are NoSchedule and PreferNoSchedule.
+	// Valid effects are NoSchedule, PreferNoSchedule and NoExecute.
 	Effect TaintEffect
+	// TimeAdded represents the time at which the taint was added.
+	// Taint's effect must be NoExecute, otherwise this field is ignored.
+	// Toleration with tolerationSeconds will tolerate the taint for
+	// only a duration (indicated with toleration.tolerationSeconds)
+	// that starts at timeAdded. By default, taint.timeAdded is not set,
+	// which means the taint can only be tolerated by toleration that
+	// tolerates infinite duration.
+	// +optional
+	TimeAdded metav1.Time
```

Comment:
Is TimeAdded only written for NoExecute taints? If so, I would mention that.
Also, if the actual implementation ignores this field (as @gmarek is suggesting), then maybe you should remove most of the text and just say "TimeAdded represents the time at which the taint was added." And if it's only written for NoExecute taint, then also say "It is only written for NoExecute taints."